main.cpp

```diff
@@ -221,18 +238,263 @@ IntType& IntType::divide(int myNumber)
     return *this;
 }
 
-int main()
+/* 
+Point implementation
+*/
+struct Point
+{
+    Point(const IntType& myNumber1, const FloatType& myNumber2);
+    Point(const IntType& myNumber1, const DoubleType& myNumber2);
+    Point(const FloatType& myNumber1, const DoubleType& myNumber2);
+
+    Point& multiply(float m);
+    Point& multiply(const IntType& myNumber);
+    Point& multiply(const FloatType& myNumber);
+    Point& multiply(const DoubleType& myNumber);
+
+    void toString();
+    
+private:
+    float x{0}, y{0};
+};
+
+Point::Point(const IntType& myNumber1, const FloatType& myNumber2) : 
+x(static_cast<float>(const_cast<IntType&>(myNumber1))), 
+y(const_cast<FloatType&>(myNumber2)) 
+{ }
+
+Point::Point(const IntType& myNumber1, const DoubleType& myNumber2) : 
+x(static_cast<float>(const_cast<IntType&>(myNumber1))), 
+y(static_cast<float>(const_cast<DoubleType&>(myNumber2))) 
+{ }
+
+Point::Point(const FloatType& myNumber1, const DoubleType& myNumber2) : 
+x(const_cast<FloatType&>(myNumber1)), 
+y(static_cast<float>(const_cast<DoubleType&>(myNumber2))) 
+{ }
```

Comment:
lol no.    Change your user-defined conversion function to work with const types and remove all of this casting.

- [ ] remove casting, fix User-Defined conversion functions on all your types.